src/components/PopoverWithMeasuredContent.js

```diff
@@ -55,12 +55,25 @@ class PopoverWithMeasuredContent extends Component {
 
         this.state = {
             isContentMeasured: false,
+            isVisible: false,
         };
 
         this.popoverWidth = 0;
         this.popoverHeight = 0;
 
         this.measurePopover = this.measurePopover.bind(this);
+        this.setContentMeasured = this.setContentMeasured.bind(this);
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        // When Popover is shown recalculate
+        if (!state.isVisible && props.isVisible) {
+            return {isContentMeasured: false, isVisible: true};
+        }
+        if (!props.isVisible) {
+            return {isVisible: false};
+        }
+        return null;
```

Comment:
I had to use it to prevent unnecessary render unmount after isVisible is set on the Popover.
we have a callback onModalHide which fires on the following conditions.
1. if Modal is visible and component unmounts.
2. Modal is set to not visible.(isVisible=false).

Previously, we are triggering the rerendering of the Modal when `isvisible` is set to true which first unmounts the old Popover and then remounts it with calculated dimensions.

> we are triggering the rerendering of the Modal when `isvisible` is set to true which first unmounts the old Popover

Here onModalHide is called due to the reason that isVisible is true and unmounting is happening.

This code does not render the popover until it is measured. we use onModalHide call to reset the reportAction and reportId back to null.

Reply:
Ok, we can maybe try to improve this in a follow up I don't have a better suggestion at the moment.

Reply:
Would be good to leave a code comment explaining this.